src/utils/eis_utils/src/eis_utils.c

```diff
@@ -701,3 +770,99 @@ EISUtilityResult RequestConnectionStringFromEISWithExpiry(
+ * @details Calls into the EIS Identity and Keyservice to create a SharedAccessSignature which is then used
+ * to create the connection string, Caller is required to call free() to deallocate the connection string
+ * @param[in] expirySecsSinceEpoch the expiration time in seconds since the epoch for the token in the connection string
+ * @param[in] timeoutMS the timeoutMS in milliseconds for each call to EIS
+ * @param[out] provisioningInfo the pointer to the struct which will be initialized with the information for creating a connection to IotHub using the EIS supported provisioning information
+ * @returns on success a null terminated connection string, otherwise NULL
+ */
+EISUtilityResult RequestConnectionStringFromEISWithExpiry(
+    const time_t expirySecsSinceEpoch, uint32_t timeoutMS, ADUC_ConnectionInfo* provisioningInfo)
+{
+    EISUtilityResult result = { EISErr_Failed, EISService_Utils };
+
+    if (provisioningInfo == NULL)
+    {
+        result.err = EISErr_InvalidArg;
+        return result;
+    }
+
+    if (expirySecsSinceEpoch <= time(NULL) || timeoutMS == 0)
+    {
+        result.err = EISErr_InvalidArg;
+        return result;
+    }
+
+    char* identityResponseStr = NULL;
+
+
+    printf("Parent process (PID %d) is about to fork a child process...\n", getpid());
+
+    // Create a named pipe (FIFO)
+    if (mkfifo(EIS_PIPE_PATH, 0666) == -1) {
+        result.err = EISErr_NamedPipeFailure;
+        goto done;
+    }
+
+    pid_t pid = fork();
+
+    if (pid < 0)
+    {
+        result.err = errno;
+        goto done;
+    }
+    else if (pid == 0)
+    {
+        // child process
+        printf("Child process (PID %d) running...\n", getpid());
+
+        if (!PermissionUtils_SetProcessEffectiveUID(EIS_USER))
+        {
+            result.err = errno;
+            goto done;
+        }
+
+        printf("Current UID calling Azure Identity Service: %d\n", getuid());
+
+        EISErr errorCode = EISIdentitiesPipeWriter(timeoutMS, EIS_PIPE_PATH);
+        if (errorCode == EISErr_Ok) {
+            exit(0);
+        } else if (errorCode == EISErr_Failed) {
+            exit(1);
+        } else {
+            exit(errorCode);
+        }
+    }
+    else
+    { // parent process
+        int status;
+        waitpid(pid, &status, 0); // wait for child process to exit
+
+        printf("Parent process (PID %d) reading data from named pipe...\n", getpid());
+
+        int childErrno;
+        EISErr EISResult = EISIdentitiesPipeReader(EIS_PIPE_PATH, &identityResponseStr, &childErrno);
+
+        if (EISResult != EISErr_Ok)
+        {
+            result.service = EISService_IdentityService;
+            result.err = EISResult;
+            errno = childErrno; // Set the errno to the value received from the child process
+            goto done;
+        }
+
+        result = ProcessIdentityResponse(provisioningInfo, identityResponseStr, expirySecsSinceEpoch, timeoutMS);
+
+    }
+done:
```

Comment:
should set ResultCode to success right before done label

Reply:
shouldn't it return whatever result ProcessIdentityResponse returns?